probe/endpoint/reporter.go

```diff
@@ -62,7 +62,7 @@ func NewReporter(hostID, hostName string, includeProcesses bool, useConntrack bo
 		}
 	}
 	if conntrackModulePresent {
-		natmapper, err = newNATMapper()
+		natmapper, err = NewNATMapper()
```

Comment:
It looks like the only time you invoke the NewNATMapper constructor is within `package endpoint`, and the only reason it's exported is so you can test it in the `endpoint_test` package. If that's true, I'm not sure it's a good idea... the API boundary of a package should be dictated by its use, and only changed to improve testability if it's necessary. But in this case, I think it's fine (better) to leave it unexported, and put its tests in `endpoint_internal_test.go` with `package endpoint`.

Not a huge deal—just something that confused me as I was reading.